src/main/java/frc/trigon/robot/subsystems/swerve/swervemodule/SwerveModule.java

```diff
@@ -11,144 +11,148 @@
+     * We save the positions to a variable instead of getting them directly because the signals update at a higher frequency than the main code loop.
+     */
+    public void update() {
         driveMotor.update();
         steerMotor.update();
         steerEncoder.update();
         latestOdometryDrivePositions = driveMotor.getThreadedSignal(TalonFXSignal.POSITION);
         latestOdometrySteerPositions = steerMotor.getThreadedSignal(TalonFXSignal.POSITION);
     }
 
-    void setDriveMotorClosedLoop(boolean closedLoop) {
-        driveMotorClosedLoop = closedLoop;
+    public void stop() {
+        driveMotor.stopMotor();
+        steerMotor.stopMotor();
     }
 
-    void setTargetState(SwerveModuleState targetState) {
-        targetState.optimize(getCurrentAngle());
-        this.targetState = targetState;
-        setTargetAngle(this.targetState.angle);
-        setTargetVelocity(this.targetState.speedMetersPerSecond, this.targetState.angle);
+    public void shouldDriveMotorUseClosedLoop(boolean shouldDriveMotorUseClosedLoop) {
+        this.shouldDriveMotorUseClosedLoop = shouldDriveMotorUseClosedLoop;
     }
 
-    /**
-     * Sets the target angle and speed of the module from a target module state for the angle, and a target current for the drive speed.
-     * This is used for PathPlanner to follow a path using feedforwards as well as chassisSpeeds.
-     *
-     * @param swerveModuleState the target module state. Only uses the angles
-     * @param targetCurrent     the target current of the drive motor
-     */
-    void setTargetState(SwerveModuleState swerveModuleState, double targetCurrent) {
-        final Rotation2d unoptimizedAngle = swerveModuleState.angle;
-        swerveModuleState.optimize(getCurrentAngle());
-        setTargetAngle(swerveModuleState.angle);
-
-        if (!swerveModuleState.angle.equals(unoptimizedAngle)) {
-            setTargetDriveMotorCurrent(-targetCurrent);
-            return;
-        }
-        setTargetDriveMotorCurrent(targetCurrent);
+    public void setDriveMotorTargetCurrent(double targetCurrent) {
+        driveMotor.setControl(driveTorqueCurrentFOCRequest.withOutput(targetCurrent));
     }
 
-    void setTargetAngle(Rotation2d angle) {
+    public void setTargetAngle(Rotation2d angle) {
         steerMotor.setControl(steerPositionRequest.withPosition(angle.getRotations()));
     }
 
-    double getDriveWheelPosition() {
-        return edu.wpi.first.math.util.Units.rotationsToRadians(driveMotor.getSignal(TalonFXSignal.POSITION));
+    public SwerveModuleState getCurrentState() {
+        return new SwerveModuleState(rotationsToMeters(driveMotor.getSignal(TalonFXSignal.VELOCITY)), getCurrentAngle());
+    }
+
+    public SwerveModuleState getTargetState() {
+        return targetState;
+    }
+
+    public Rotation2d getDriveWheelPosition() {
+        return Rotation2d.fromRotations(driveMotor.getSignal(TalonFXSignal.POSITION));
```

Comment:
This was in radians because it was needed for the wheel radius characterization. Normally we don't use angle measurements as distance but because it is done like that in this case, we use radians instead of a Rotation2d because a Rotation2d symbolizes a rotation.